homework/queries.sql

```diff
@@ -1,19 +1,93 @@
 --- Get the top 3 product types that have proven most profitable
-
+SELECT product_line, SUM(profit) AS product_profit
+FROM products
+INNER JOIN profit
+ON products.product_code = profit.product_code
+GROUP BY product_line
+ORDER BY product_profit DESC
+LIMIT(3);
 
 --- Get the top 3 products by most items sold
-
+SELECT product_name, SUM(quantity_ordered) AS items_sold
+FROM products
+INNER JOIN profit
+ON products.product_code = profit.product_code
+GROUP BY products.product_code
+ORDER BY items_sold DESC
+LIMIT(3);
 
 --- Get the top 3 products by items sold per country of customer for: USA, Spain, Belgium
+(SELECT product_name, customers.country, SUM(quantity_ordered) AS items_sold
+FROM products
+INNER JOIN profit
+ON products.product_code = profit.product_code
+INNER JOIN customers
+ON profit.customer_number = customers.customer_number
+WHERE customers.country = 'USA'
+GROUP BY customers.country, product_name
+ORDER BY items_sold DESC
+LIMIT(3))
+UNION ALL
+(SELECT product_name, customers.country, SUM(quantity_ordered) AS items_sold
+FROM products
+INNER JOIN profit
+ON products.product_code = profit.product_code
+INNER JOIN customers
+ON profit.customer_number = customers.customer_number
+WHERE customers.country = 'Spain'
+GROUP BY customers.country, product_name
+ORDER BY items_sold DESC
+LIMIT(3))
+UNION ALL
+(SELECT product_name, customers.country, SUM(quantity_ordered) AS items_sold
+FROM products
+INNER JOIN profit
+ON products.product_code = profit.product_code
+INNER JOIN customers
+ON profit.customer_number = customers.customer_number
+WHERE customers.country = 'Belgium'
+GROUP BY customers.country, product_name
+ORDER BY items_sold DESC
+LIMIT(3));
 
 
 --- Get the most profitable day of the week
-
+SELECT date_weekday, SUM(profit) AS total_profit
+FROM dates
+INNER JOIN profit
+ON dates.full_date = profit.order_date
+GROUP BY date_weekday
+ORDER BY total_profit DESC
+LIMIT(1);
 
 --- Get the top 3 city-quarters with the highest average profit margin in their sales
+SELECT offices.city, dates.date_quarter, AVG(profit_margin) AS avg_margin
+FROM dates
+INNER JOIN profit
+ON dates.full_date = profit.order_date
+INNER JOIN offices
+ON profit.office_code = offices.office_code
+GROUP BY offices.city, dates.date_quarter
+ORDER BY avg_margin DESC
+LIMIT(3);
 
 
--- List the employees who have sold more goods (in $ amount) than the average employee.
+--- List the employees who have sold more goods (in $ amount) than the average employee.
+SELECT employees.employee_number AS top_employees, SUM(revenue) AS total_revenue
```

Comment:
It's fine, but do you notice yourself asking the database to repeat the same operations twice?